jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile

```diff
@@ -0,0 +1,158 @@
+                dir("rhmi-jenkins") {
+                    git branch: "master", url: "https://gitlab.cee.redhat.com/integreatly-qe/rhmi-jenkins.git"
+                    def dirExists = fileExists(clusterInfoDirectory)
+                    if (dirExists) {
+                        error "[ERROR] Configuration files found for ${clusterName}! Make sure that the cluster name is unique"
+                    } 
+                    sh "mkdir -p ${clusterInfoDirectory}"
+                }
+            } // stage
+            
+            stage("Update the cluster creation template information") {
+                dir("rhmi-jenkins") {
+                    sh(
+                        returnStdout: false,
+                        script: """
+                            cp templates/cluster.json ${clusterInfoDirectory}
+                            sed -i 's/CLUSTER_NAME/${clusterName}/g' ${clusterInfoDirectory}/cluster.json
+                            echo 'Creating OpenShift 4 OSD cluster with the following configuration:'
+                            cat ${clusterInfoDirectory}/cluster.json
+                        """
+                    )
+                }
+            } // stage
+
+            stage("login to ocm") {
+                withCredentials([string(credentialsId: "${OCM_TOKEN}", variable: "OCM_ACCESS_TOKEN")]) {
+                    sh "ocm login --url=https://api.stage.openshift.com/ --token=${OCM_ACCESS_TOKEN}"
+                }
+            } // stage
+
+            stage("Push the cluster template to GitLab") {
+                withCredentials([string(credentialsId: "gitlab-access-token", variable: "GITLAB_ACCESS_TOKEN")]) {
+                    dir("rhmi-jenkins") {
+                        sh(
+                            returnStdout: false,
+                            script: """
+                                git remote add jenkins https://oauth2:${GITLAB_ACCESS_TOKEN}@gitlab.cee.redhat.com/integreatly-qe/rhmi-jenkins.git | true
+                                git config --global user.email "integreatly-qe@redhat.com" | true
+                                git config --global user.name "rhmi-jenkins-bot" | true
+                                git add -A
+                                git commit -m 'cluster template created: ${clusterName}'
+                                git push jenkins master
+                            """
+                        )
+                    }
+                }
+            } // stage
+            
+            stage("Begin cluster creation") {
+                rootWorkspaceDir = sh (
+                    script: "pwd",
+                    returnStdout: true
+                ).trim()
+                installConfigDirectory = rootWorkspaceDir + "/rhmi-jenkins/" + clusterInfoDirectory + "/"
+                sh "ocm post /api/clusters_mgmt/v1/clusters --body=${installConfigDirectory}cluster.json > ${clusterInfoFileName}"
```

Comment:
Maybe the `/api/clusters_mgmt/v1/clusters` is a good candidate for a constant.

Reply:
sure. makes sense

Reply:
@tremes I think it is ok now, but there seems to be some issue with the cluster creation infrastructure, therefore the cluster was not provisioned within 2 hours (still provisioning), while it usually takes approx 30 minutes